examples/with-emotion/pages/_document.js

```diff
@@ -2,21 +2,27 @@ import Document, { Head, Main, NextScript } from 'next/document'
 import { extractCritical } from 'emotion-server'
 
 export default class MyDocument extends Document {
-  static getInitialProps({ renderPage }) {
-    const page = renderPage()
-    const styles = extractCritical(page.html)
-    return { ...page, ...styles }
+  static async getInitialProps(ctx) {
+    const initialProps = await Document.getInitialProps(ctx)
+    const styles = extractCritical(initialProps.html)
```

Comment:
What's the point of using `extractCritical` at all? I've used emotion for quite a while with Next and never had to do that.

Reply:
@Vadorequest See https://emotion.sh/docs/ssr

Reply:
> This only applies if you’re using vanilla Emotion or a version of Emotion prior to v10. For v10 and above, SSR just works in Next.js.

![image](https://user-images.githubusercontent.com/3807458/83136068-23660700-a0e7-11ea-877b-03bd3e755786.png)

https://emotion.sh/docs/ssr#nextjs

I believe you were mislead, and what this example showcases isn't necessary.
There doesn't seem to be anything needed to be done when using Next.js.

Here is (for comparison) my own setup:
- Source code `_document` https://github.com/UnlyEd/next-right-now/blob/v1-hyb-mst-aptd-gcms-lcz-sty/src/pages/_document.tsx#L55
- Live demo https://nrn-v1-hyb-mst-aptd-gcms-lcz-sty-c1-i6tj62ina.now.sh/

Maybe I'm missing something? But it works great without the above complexity.

Reply:
Like the documentation says, Emotion has two ways of doing SSR. You're using the "default approach" and I'm using the "advanced approach". Both work just fine.

Reply:
Alright, but, what's the benefit of using the advanced approach?

I mean, if both have the same benefit, why would you recommend newcomers to face something that's actually harder to configure for no benefit whatsoever? 🤔

Reply:
Since Next doesn't support streaming, the only benefit of the "default approach" is that it works out of the box. It doesn't support certain selectors and style tags outside of head aren't valid HTML and may cause other issues (https://github.com/whatwg/html/issues/1605#issuecomment-235956455 and https://github.com/whatwg/html/issues/1605#issuecomment-277826129).

Having said that, I'm not recommending one approach over the other. I just updated an already existing example.

Reply:
Thank you very much for the clarification, I wasn't aware of those limitations.